service/src/main/java/bio/terra/workspace/app/configuration/spring/LandingZoneStairwayConfiguration.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.workspace.app.configuration.spring;
+
+import bio.terra.common.kubernetes.KubeProperties;
+import bio.terra.common.kubernetes.KubeService;
+import bio.terra.common.stairway.StairwayComponent;
+import bio.terra.common.stairway.StairwayProperties;
+import bio.terra.landingzone.library.configuration.LandingZoneAzureConfiguration;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
```

Comment:
Sure. I did it on purpose thinking that library should be initialized from the code which uses it. But since library is fully autonomous and yes it has DB configuration inside it make sense to move the configuration into library.

Reply:
Now there are two Stairway components. One serves WSM and another one is for Landing Zone library. I added separate configuration for LZ Stairway component so we can adjust them separately if needed. There is a StairwayProperties component defined in terra common lib. In LZ library I am using Qualifiers. But in terra common lib there are no markers for Spring which help to resolve ambiguity. We probably need to mark StairwayProperties in common lib as Primary to remove this ambiguity. Am I missing something?

Reply:
Ah, that is a problem. The TCL implementation is assuming one Stairway per service.
If you think adding `@Primary` in TCL would help, I would do that.

Another alternative would be to take properties as input in the StairwayOptionsBuilder, having them default to the `@Autowired` properties. Then you could selectively override.

I suppose another alternative is to share a single Stairway instance.

Reply:
I will try to add `@Primary` and see how it goes..

Reply:
I believe another approach could be to wire up an instance of the `LandingZoneService` in the configuration and avoid having two instances of the `StairwayComponent` in the app context. 

In the LZ service:
 - Remove the `@Component` tag from the `LandingZoneService` class.

In WSM (or whatever location makes sense to have the configuration):
 - Have something along these lines:
```java
  @Bean
  public LandingZoneService getLandingZoneService( LandingZoneJobConfiguration jobConfig,
                                                   LandingZoneIngressConfiguration ingressConfig,
                                                   LandingZoneStairwayDatabaseConfiguration stairwayDatabaseConfiguration,
                                                   LandingZoneMdcHook mdcHook,
                                                   LandingZoneFlightBeanBag flightBeanBag,
                                                   KubeService kubeService,
                                                   KubeProperties kubeProperties,
                                                   @Qualifier("landingZoneStairwayProperties") StairwayProperties stairwayProperties,
                                                   ObjectMapper objectMapper){

    StairwayComponent stairwayComponent = new StairwayComponent(kubeService, kubeProperties, stairwayProperties);

    return new  LandingZoneJobService(
            LandingZoneJobConfiguration jobConfig,
            LandingZoneIngressConfiguration ingressConfig,
            LandingZoneStairwayDatabaseConfiguration stairwayDatabaseConfiguration,
            LandingZoneMdcHook mdcHook,
            @Qualifier("landingZoneStairwayComponent") StairwayComponent stairwayComponent,
            LandingZoneFlightBeanBag flightBeanBag,
            ObjectMapper objectMapper)
  }
```

In my suggestion I am assuming that the only conflicting component is `StairwareComponent`, if other components can't be duplicated in the app context, the same pattern applies: create instances explicitly in the method body and only expect shared instances as parameters of `getLandingZoneService`.

Reply:
But the problem is that Spring can't construct StairwayComponent, because we have 2 StairwayProperties objects. One for WSM, the other one for landing zone.

Reply:
Seems we want to get rid of configuration. The problem is that Spring couldn't construct StairwayComponent because of 2 Stairway configuration. Probably this configuration shouldn't be enough.

Reply:
Can you just wrap `StairwayProperties` in another type or create a new type all together with the required values -`LandingZoneStairwayProperties`? This way you can have Spring inject this instance and construct your local instance of the `StairwayComponent` from it.

Reply:
@ddietterich, I'd appreciate your thoughts on this approach which boils down to having two instances of `StairwayComponent` running.  Do you see any issues?  My concern is from your comment above that the TCL implementation assumes one stairway per service. Does the assumption apply only to the instantiation of the components (which we are trying to work around), or does it have negative implications at runtime?

Reply:
@giventocode So, I made a wrapper around StairwayProperties. Move configuration into lz library and remove it from wsm. Tested it locally and it looks good. Waiting for green status here.

Reply:
I deleted configuration from WSM and moved it into LZ library. Also I wrapped StairwayProperties to remove ambiguity. LandingZone PR in is [opened](https://github.com/DataBiosphere/terra-landing-zone-service/pull/16)

Reply:
It is fine to have multiple Stairways running within the service. And that serves LZ well, because you can test the flights in the library without needing WSM.

I don't like using Spring magic for the multi-stairway case. It is difficult to see what is going on. Eventually, I would  like to see a change to TCL to take properties as a parameter instead of (or in addition to) the autowiring of the default properties.